web/src/main/webapp/my-app/layout/controllers.js

```diff
@@ -47,11 +47,127 @@ define(['angular', 'jquery'], function(angular, $) {
     function($localStorage, $log, $sessionStorage,
       $scope, $rootScope, layoutService) {
       var vm = this;
+
+      /**
+       * Log whenever a widget is moved
+       * @param eventType {String} Kind of interaction that moved the widget
+       * @param fname {String} The moved widget's fname
+       * @param dropIndex {Number} Index where the widget landed
+       * @param startIndex {Number} (optional) Index before moving the widget
+       * @return {boolean}
+       */
+      $scope.logEvent = function(eventType, fname, dropIndex, startIndex) {
+        switch (eventType) {
+          case 'dragEnd':
+            $log.info('Dragged ' + fname + ' to index ' + dropIndex);
+            break;
+          case 'keyboardMove':
+            $log.info('Moved ' + fname + 'from index ' + startIndex +
+              ' to index ' + dropIndex);
+            break;
+          default:
+            return true;
+        }
+      };
+
+      /**
+       * Respond to arrow key-presses when focusing a movable list element
+       * @param widget {Object} The widget trying to move
+       * @param event {Object} The event object
+       * @return {boolean} True if proper conditions aren't met
```

Comment:
What's up with the `boolean` return? `true` means didn't do anything? Is this idiomatic? In some Java methods, for instance, `true` means had an effect and `false` means didn't have an effect, e.g. [Set.add()](https://docs.oracle.com/javase/7/docs/api/java/util/Set.html#add(E)) .

Reply:
Agree it's semantically meaningless, just the result of hasty writing. This is something I can clean up!